### PR TITLE
Test all common and complex data types (minus Nested)

### PR DESCRIPTION
**Pull request by slvrtrn**
#55 

Insert & Select
--
- [x] A dataset of several rows with all the language primitives
- [x] A test suit for each supported ClickHouse Data Type (Enum, Map, UInt 256, etc.)

`Nested` is skipped for the time being. 
`Decimal` does not work with the `JSON*` input format, not sure if it is intended or not.

### CODE REVIEW

File: __tests__/integration/schema_types.test.ts [hunk 1]
```diff
@@ -383,12 +383,6 @@ async function assertInsertAndSelect<S extends ch.Shape>(
   await table.insert({
     values: value,
   })
-  const result = await (
-    await table.select({
-      clickhouse_settings: {
-        s3_create_new_file_on_insert: 1,
```
Comment (slvrtrn) [inline __tests__/integration/schema_types.test.ts:389]: `¯\_(ツ)_/¯`

File: __tests__/integration/data_types.test.ts [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+      {
+        'n.id': 43,
+        'n.name': 'bar',
+        'n.createdAt': '2000-01-12',
+        'n.roles': ['Admin'],
+      },
+    ])
+  })
+
+  // TODO should belong to fixtures as it is quite useful in other tests
```
Comment (slvrtrn) [inline __tests__/integration/data_types.test.ts:438]: will move in the next one

File: __tests__/integration/data_types.test.ts [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+  //  -     "n.name": "foo",
+  //  -     "n.roles": Array [
+  //  -       "User",
+  //  -     ],
+  //  +     "n.createdAt": Array [],
+  //  +     "n.id": Array [],
+  //  +     "n.name": Array [],
+  //  +     "n.roles": Array [],
+  //    },
+  it.skip('should work with nested', async () => {
```
Comment (slvrtrn) [inline __tests__/integration/data_types.test.ts:386]: I genuinely have no idea how to insert and query it properly after spending some time on the test.
Neither nested objects like below nor fields like `n.id` worked out. Noticed an interesting thing though: to get field names like `n.id` to work, you need to manually escape it:

```
{
  '`n.id`': 42,
  '`n.name`': 'foo',
  '`n.createdAt`': '2001-04-23',
  '`n.roles`': ['User'],
}
```
as our HTTP interface does not like the identifiers with dots.

File: __tests__/integration/data_types.test.ts [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+    // FIXME (?) does not work with JSON* insert formats, only raw streaming
+    //   as decimals are required to be passed as numbers
+    //   when using JSONEachRow (why?)
+    it('should work with decimals', async () => {
+      const stream = new Stream.Readable({
+        objectMode: false,
+        read() {
+          //
+        },
+      })
```
Comment (slvrtrn) [inline __tests__/integration/data_types.test.ts:105]: Stream is required to bypass the `encode` function which only works with `JSON*` currently.
This is a very tricky case.

https://github.com/ClickHouse/clickhouse-js/blob/7e29e4f4c6b85923eec588e828fbb897bcfec41e/src/data_formatter/formatter.ts#L88-L99

File: __tests__/integration/data_types.test.ts [hunk 5]
```diff
@@ -0,0 +1,488 @@
...
+      ]
+      const table = await createTableWithFields('ip1 IPv4, ip2 IPv6')
+      await insertAndAssert(table, values)
+    })
+
+    it('should work with ((very) nested) arrays', async () => {
+      // it's the largest reasonable nesting value (data is generated within 50 ms);
+      // 25 here can already tank the performance to ~500ms only to generate the data;
+      // 50 simply times out :)
+      const maxNestingLevel = 20
```
Comment (slvrtrn) [inline __tests__/integration/data_types.test.ts:234]: I could not resist testing such an edge case (also for maps)
